persist-cli/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -134,17 +144,74 @@ public void execute() {
             try {
                 BalProjectUtils.validateSchemaFile(file);
                 entityModule = BalProjectUtils.getEntities(file);
+                String fileName = entityModule.getModuleName();
+                if (!ProjectUtils.validateModuleName(fileName)) {
+                    errStream.println("ERROR: invalid definition file name : '" + fileName + "' :\n" +
+                            "file name can only contain alphanumerics, underscores and periods");
+                    return;
+                } else if (!ProjectUtils.validateNameLength(fileName)) {
+                    errStream.println("ERROR: invalid definition file name : '" + fileName + "' :\n" +
+                            "maximum length of module name is 256 characters");
+                    return;
+                }
+                generatedSourceDirPath = Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY);
+                HashMap<String, String> persistConfig = readPersistConfigurations(
+                        Paths.get(this.sourcePath, "Ballerina.toml"));
+
+                if (!persistConfig.get("module").equals(packageName)) {
+                    String submodule = persistConfig.get("module").split("\\.")[1];
+                    if (!ProjectUtils.validateModuleName(submodule)) {
+                        errStream.println("ERROR: invalid module name : '" + submodule + "' :\n" +
+                                "module name can only contain alphanumerics, underscores and periods");
+                        return;
+                    } else if (!ProjectUtils.validateNameLength(submodule)) {
+                        errStream.println("ERROR: invalid module name : '" + submodule + "' :\n" +
+                                "maximum length of module name is 256 characters");
+                        return;
+                    }
+                    generatedSourceDirPath = Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY,
+                            submodule);
+                }
+                if (!Files.exists(generatedSourceDirPath)) {
+                    try {
+                        if (!persistConfig.get("module").equals(packageName) && !Files.exists(
+                                Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY))) {
+                            Files.createDirectory(Paths.get(this.sourcePath,
+                                    BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY).toAbsolutePath());
+
+                        }
```

Comment:
Do you need this logic here?

Reply:
yes, otherwise it can give errors when the generated directory is not there

Reply:
I mean, we can simplify like this, as we already have `generatedSourceDirPath` 

```
if (!Files.exists(generatedSourceDirPath)) {
    try {
       Files.createDirectories(generatedSourceDirPath.toAbsolutePath());
    } catch (IOException e) {
        errStream.println("ERROR: failed to create the generated directory. " + e.getMessage());
        return;
    }
}
```